src/configure/templateInputHelper/InputControlProvider.ts

```diff
@@ -28,14 +29,31 @@ export class InputControlProvider {
         this._createControls();
     }
 
+    public async overrideParameters(inputControl: InputControl) {
+        let properties = inputControl.getPropertyValue(constants.clientPropertyKey);
+        let arrayofProperties = Object.keys(properties);
+
+        arrayofProperties.forEach(element => {
+            let key = element.split(".", 2)[1];
+            let dependentInputControlArray = this._getInputDependencyArray(inputControl, [properties[element]], false);
+            let dependentClientInputMap = this._getClientDependencyMap(inputControl, [properties[element]]);
+            let newValue = this._computeMustacheValue(properties[element], dependentInputControlArray, dependentClientInputMap);
+            inputControl.updateInputDescriptorProperty(key, newValue);
+            if (key === constants.inputModeProperty) {
+                let updatedControlType = InputControlUtility.getInputControlType(parseInt(newValue));
+                inputControl.updateControlType(updatedControlType);
+            }
+        });
+    }
+
     public async getAllPipelineTemplateInputs() {
         let parameters: { [key: string]: any } = {};
         for (let inputControl of this._inputControlsMap.values()) {
             if (this.repoAnalysisSettingInputProvider.inputFromRepoAnalysisSetting(inputControl)) {
                 await this.repoAnalysisSettingInputProvider.setInputControlValueFromRepoAnalysisResult(inputControl);
             }
-            else if (inputControl.getPropertyValue('deployTarget') === "true" && !!this._context["resourceId"]) {
-                inputControl.setValue(this._context["resourceId"]);
+            else if (!!inputControl.getPropertyValue(constants.clientPropertyKey)) {
```

Comment:
logic here should be first override parameters of inputControl and then set its dataSourceControls, initializaDynamicValidations. After this set the value from repoAnalysis or else condition.
code in for loop should be like:
(!!inputControl.getPropertyValue(constants.clientPropertyKey)) {
   this.overrideParameters(inputControl);
}
this._setInputControlDataSourceInputs(inputControl);
this._initializeDynamicValidations(inputControl);
 if (this.repoAnalysisSettingInputProvider.inputFromRepoAnalysisSetting(inputControl)) {
                await this.repoAnalysisSettingInputProvider.setInputControlValueFromRepoAnalysisResult(inputControl);	                } else {
    this._setInputControlVisibility(inputControl);
    this._setupInputControlDefaultValue(inputControl);
     await inputControl.setInputControlValue();
}

Reply:
Updated